modules/reindex/src/test/java/org/elasticsearch/index/reindex/RestDeleteByQueryActionTests.java

```diff
@@ -21,20 +21,39 @@
 
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
-import org.elasticsearch.rest.RestController;
-import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.search.RestSearchAction;
 import org.elasticsearch.test.rest.FakeRestRequest;
+import org.elasticsearch.test.rest.RestActionTestCase;
 
+import org.junit.Before;
 import java.io.IOException;
 
 import static java.util.Collections.emptyList;
-import static org.mockito.Mockito.mock;
 
-public class RestDeleteByQueryActionTests extends ESTestCase {
+public class RestDeleteByQueryActionTests extends RestActionTestCase {
+    private RestDeleteByQueryAction action;
+
+    @Before
+    public void setUpAction() {
+        action = new RestDeleteByQueryAction(Settings.EMPTY, controller());
+    }
+
+    public void testTypeInPath() {
+        RestRequest request = new FakeRestRequest.Builder(xContentRegistry())
+            .withMethod(RestRequest.Method.POST)
+            .withPath("/some_index/some_type/_delete_by_query")
+            .build();
+        dispatchRequest(request);
+        // checks the type in the URL is propagated correctly to the request object
+        assertEquals("some_type", request.param("type"));
```

Comment:
I was thinking we could have a separate test for request building where we call `action.buildRequest(...)` and check that the `DeleteByQueryRequest` contains the right document types.